File: 2022 test/day1/P2.py
file = open("day2/input.txt", "r")
max=[0, 0, 0]; nombre=0; somme=0
finLutin = False
while True:
    c = file.read(1)
    if c=="":
        break
    if c!="\n":
        finLutin = False
        nombre = 10*nombre + int(c)
    if c=="\n":
        if finLutin == True:
            if somme > max[0] :
                if somme > max[1]:
                    if somme > max[2]:
                        max[0] = max[1]
                        max[1] = max[2]
                        max[2] = somme
                    else:
                        max[0] = max[1]
                        max[1] = somme
                else:
                    max[0] = somme
            somme = 0
        somme+=nombre
        nombre = 0
        finLutin = True
print(max[0]+max[1]+max[2])

File: 2021/day6/P2.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>
#include <iterator>
#include <unordered_map>
#include <cmath>

using namespace std;

unordered_map<int, long long int> getInput(const char *filename)
{
    unordered_map<int, long long int> result;

    ifstream input_stream(filename);
    if (!input_stream)
        cerr << "Can't open input file!";

    string line;
    while (getline(input_stream, line, ','))
    {
        int internal_timer = stoi(line);
        result[internal_timer] += 1;
    }

    return result;
}

int main()
{
    unordered_map<int, long long int> lanternfishs = getInput("2021/day6/input.txt");

    for (int day=0; day < 256; day++)
    {
        long long int new_lanternfishs = lanternfishs[0]; // Sauvegarde du nombre de lanternfish qui s'apprètent à accoucher
        // Update de tous les internal timer des lanternfish
        for (int internal_timer=0; internal_timer<8; internal_timer++)
        {
            lanternfishs[internal_timer] = lanternfishs[internal_timer+1];
        }
        // Naissance des nouveaux lanternfishs + reset des mamans
        lanternfishs[8] = new_lanternfishs;
        lanternfishs[6] += new_lanternfishs;
    }

    // Calcul de la somme de tous les lanternfishs au bout de 80 jours
    long long int sum = 0;
    for (int i=0; i<9; i++)
    {
        sum += lanternfishs[i];
    }
    cout << sum << endl;
}